cmd/livepeer_bench/livepeer_bench.go

```diff
@@ -0,0 +1,216 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	//"runtime/pprof"
+
+	"github.com/livepeer/go-livepeer/common"
+	"github.com/livepeer/lpms/ffmpeg"
+	"github.com/livepeer/m3u8"
+)
+
+func main() {
+	/*
+		cprof, err := os.Create("bench.prof")
+		if err != nil {
+			panic(err)
+		}
+		pprof.StartCPUProfile(cprof)
+		defer pprof.StopCPUProfile()
+	*/
+	// Override the default flag set since there are dependencies that
+	// incorrectly add their own flags (specifically, due to the 'testing'
+	// package being linked)
+	flag.Set("logtostderr", "true")
+	flag.CommandLine = flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+
+	fname := flag.String("fname", "", "Input m3u8 manifest file")
+	conc := flag.Int("conc", 1, "# of concurrent transcode sessions")
+	segs := flag.Int("segs", 0, "Maximum # of segments to transcode (default all)")
+	profs := flag.String("profs", "P240p30fps16x9,P360p30fps16x9,P720p30fps16x9", "Transcoding options for broadcast job, or path to json config")
+	nvidia := flag.String("nvidia", "", "Comma-separated list of Nvidia GPU device IDs to use for transcoding")
+	outPrefix := flag.String("outPrefix", "", "Output segments' prefix (no segments are generated by default)")
+
+	flag.Parse()
+
+	if *fname == "" {
+		panic("Please provide the input manifest as `-fname <input.m3u8>`. See -h or -help for more.")
+	}
+
+	profiles := parseVideoProfiles(*profs)
+
+	f, err := os.Open(*fname)
+	if err != nil {
+		panic(err)
+	}
+	p, _, err := m3u8.DecodeFrom(bufio.NewReader(f), true)
+	if err != nil {
+		panic(err)
+	}
+	pl, ok := p.(*m3u8.MediaPlaylist)
+	if !ok {
+		panic("Expecting media PL")
+	}
+
+	accel := ffmpeg.Software
+	devices := []string{}
+	if *nvidia != "" {
+		accel = ffmpeg.Nvidia
+		devices = strings.Split(*nvidia, ",")
+	}
+
+	ffmpeg.InitFFmpeg()
+	segCount := 0
+	var wg sync.WaitGroup
+	dir := path.Dir(*fname)
+	start := time.Now()
+	fmt.Fprintf(os.Stderr, "Program %s Source %s Concurrency %d Profiles %s\n", os.Args[0], *fname, *conc, *profs)
```

Comment:
> WDYT of printing this to os.Stdout instead of os.Stderr as it seems informative rather than an error ?

I liked how the tool printed stats to `stdout` and rest of the information on `stderr` - this allows the user to do a quick export of the stats JSON `./livepeer_bench ... > stats.json` while still seeing the other information on their terminal output.

Ofc this can be handled better by taking in the JSON filename as an optional argument - but for now this seemed good enough and unix-like. WDYT?

> Alternatively what do you think of displaying this in a table format ? For example

Yeah that looks really cool, I'll try it and send an update.

Reply:
s/json/csv

Reply:
Agree with @jailuthra here on the value of being able to export the stats in a standard format such as CSV so that the data can be easily imported into other tools. I suggest we stick with the current formatting and consider improvements such as different stats visualization formats outside of this PR.